script.skin.helper.service/resources/lib/listitem_monitor.py

```diff
@@ -0,0 +1,585 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+'''
+    script.skin.helper.service
+    listitem_monitor.py
+    monitor the kodi listitems and providing additional information
+'''
+
+import threading
+import thread
+from utils import log_msg, log_exception, get_current_content_type, kodi_json, prepare_win_props
+from metadatautils import extend_dict, process_method_on_list
+import xbmc
+from simplecache import SimpleCache
+
+
+class ListItemMonitor(threading.Thread):
+    '''Our main class monitoring the kodi listitems and providing additional information'''
+    event = None
+    exit = False
+    delayed_task_interval = 1795
+    listitem_details = {}
+    all_window_props = []
+    cur_listitem = ""
+    last_folder = ""
+    last_listitem = ""
+    foldercontent = {}
+    screensaver_setting = None
+    screensaver_disabled = False
+    lookup_busy = {}
+    enable_extendedart = False
+    enable_musicart = False
+    enable_animatedart = False
+    enable_extrafanart = False
+    enable_pvrart = False
+    enable_forcedviews = False
+    bgtasks = 0
+
+    def __init__(self, *args, **kwargs):
+        self.cache = SimpleCache()
+        self.metadatautils = kwargs.get("metadatautils")
+        self.win = kwargs.get("win")
+        self.kodimonitor = kwargs.get("monitor")
+        self.event = threading.Event()
+        threading.Thread.__init__(self, *args)
+
+    def stop(self):
+        '''called when the thread has to stop working'''
+        log_msg("ListItemMonitor - stop called")
+        self.exit = True
+        self.event.set()
+
+    def run(self):
+        '''our main loop monitoring the listitem and folderpath changes'''
+        log_msg("ListItemMonitor - started")
+        self.get_settings()
+
+        while not self.exit:
```

Comment:
well, that check is alread performed by the main_service which controls the additional threads. Once an abort is requested by Kodi, the exit flag will be set.